numpy/core/src/umath/loops_arithm_fp.dispatch.c.src

```diff
@@ -332,13 +336,22 @@ sse2_binary_scalar2_@kind@_@TYPE@(@type@ * op, @type@ * ip1, @type@ * ip2, npy_i
     }
 #else
     const npy_intp vector_size_bytes = 16;
-    const @vtype@ b = @vpre@_set1_@vsuf@(ip2[0]);
+    #if @FASTDIV@ // XXX Implement other bit sizes
```

Comment:
Try to not be affected by our legacy code, here is an example for SIMD kernel that only supports contiguous memory access.
```c

#if NPY_SIMD
/**begin repeat
 * SFX = s8, s16, s32#
 */
static void simd_divide_by_scalar_contig_contig_@sfx@
(const npyv_lanetype_@sfx@ *src, npyv_lanetype_@sfx@ scalar, const npyv_lanetype_@sfx@ *dst)
{  
    const int vstep = npyv_nlanes_@sfx@;
    const npyv_@sfx@x3 divisor = npyv_divisor_@sfx@(scalar);

    for (; len >= vstep; len -= vstep, src += vstep, dst += vstep) {
        npyv_@sfx@ a = npyv_load_@sfx@(src);
        npyv_@sfx@ c = npyv_divc_@sfx@(a, divisor);
        npyv_store_@sfx@(dst, c);
    }
    for (; len > 0; --len, ++src, ++dst) {
        const npyv_lanetype_@sfx@ a = *src;
        *dst = a / scalar;
    }
    npyv_cleanup();
}
 /**end repeat**/ 
#endif
````
We can support non-contiguous memory access in another pull-request, let's keep it simple for now.